var/spack/repos/builtin/packages/rivet/package.py

```diff
@@ -72,8 +75,15 @@ class Rivet(AutotoolsPackage):
 
     depends_on("hepmc", when="hepmc=2")
     depends_on("hepmc3", when="hepmc=3")
+    # The fix for working around patch-level zero issues have landed in 4.0.1
+    # and have also been back-ported to 3.1.11. Hence we need to exclude 4.0.0
+    # explicitly and catch the rest with versions up to 3.1.10
+    # See: https://gitlab.com/hepcedar/rivet/-/merge_requests/904
+    # and: https://gitlab.com/hepcedar/rivet/-/merge_requests/912
     conflicts(
-        "hepmc@3.3.0", when="@:4.0.0 hepmc=3", msg="patch-level zero requires at least 4.0.1"
+        "hepmc@3.3.0",
+        when="@:3.1.10,4.0.0 hepmc=3",
+        msg="patch-level zero requires at least 3.1.11 or 4.0.1",
```

Comment:
Message allows ambiguity about which patch-level zero we're talking about.
```suggestion
        msg="HepMC 3.3 patch-level zero requires at least 3.1.11 or 4.0.1",
```

Reply:
The issue was technically all dependency versions with patch level 0, not just HepMC 3.3.0. However, I think HepMC 3.3.0 was the first one that actually triggered this (presumably because there were no X.Y.0 release before, I actually don't know). I am not sure there is an easy way to express that fact though.

However, I am fine with adjusting the message here to be HepMC 3.3.0 specific since that will most likely be the one that people stumble over.

Reply:
Also happy to keep it just HepMC, without version, or:
```suggestion
        msg="HepMC 3.x.0 requires at least 3.1.11 or 4.0.1",
```
And then we could even have the conflict apply to `^hepmc@3.1.0,3.2.0,3.3.0`.

Reply:
Done (including the extended version conflict)